test/base58_test.cpp

```diff
@@ -0,0 +1,44 @@
+#include <gtest/gtest.h>
+#include <vector>
+#include <veriblock/base58.h>
+
+
+TEST(base58, Encode)
+{
+     //  test data
+     std::vector<uint8_t> inputA = { 27,53,84,6,78,51,29,38,57,2,0,95,23,89,4,6,7,92,35,61,30,95,12,96,29 };
+     std::vector<uint8_t> inputB = { 2,9,5,7,9,3,4,5,7,3,8,9,4,6,7,3,7,8,9,6,7,9,5,7,2,9,8,5,2,9,8,5,7,2 };
+     std::vector<uint8_t> inputC = { 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 };
+     std::vector<uint8_t> inputD = { };
```

Comment:
this looks like separate test cases.
much better would be to create a test case
```
struct TestCase {
  std::vector<uint8_t> input;
  std::string output;
};
```

And create 2 value-parametrized tests:
- Encode: take `input`, encode it, compare it to `output`
- Decode: take `output`, decode it, compare it to `input`

Reply:
fixed